aws/resource_aws_elasticache_replication_group.go

```diff
@@ -102,6 +102,20 @@ func resourceAwsElasticacheReplicationGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					versions := strings.Split(new, ".")
+
+					// For example 6.x
+					if len(versions) == 2 {
+						major, minor := versions[0], versions[1]
+
+						if minor == "x" && strings.HasPrefix(old, major) {
```

Comment:
What if someone created Redis with `"5.0.5"` and then later changed it to `"5.x"`? I wouldn't expect someone to actually do that, but if someone did, rather than getting an expected apply-time error, your `DiffSuppressFunc` might actually cause TF ignore that change if I follow your code correctly.

I think what you want to do is to compare the old major version and the new major version, check that it's >= 6, then return `true` if the both major versions are equal and the new minor version == "x".

I.e. `6.0.5 -> 6.x` - the major versions are equal, the major version is >=6, and new minor version == "x". ✅